### PR TITLE
JSON: write struct array nulls as null

### LINKED ISSUE #5066: JSON writer - write null structs as null, instead of struct with nulls
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

See note here:

https://github.com/apache/arrow-rs/blob/31b5724332666d68fe94a5b3572a13a51022ea81/arrow-json/src/writer.rs#L1315-L1328

**Describe the solution you'd like**

When writing a struct array, if there is a null then should write null instead of `{}`

**Describe alternatives you've considered**

Keep behaviour as is, writing `{}` for null entry in struct array. Remove above note in code and clearly document this is expected behaviour.

**Additional context**

### PR DESCRIPTION
**Pull request by Jefffrey**
# Which issue does this PR close?

Closes #5066

# Rationale for this change
 

# What changes are included in this PR?

Nulls in struct arrays are written as `null` in JSON, where previously was written as `{}` (or filled with nulls for the keys if explicit nulls are written)

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-json/src/writer.rs [hunk 1]
```diff
@@ -129,12 +129,20 @@ where
...
-        .collect::<Vec<JsonMap<String, Value>>>();
+    let mut inner_objs = (0..array.len())
+        // Ensure we write nulls for struct arrays as nulls in JSON
+        // Instead of writing a struct with nulls
+        .map(|index| {
+            if array.is_null(index) {
+                None
+            } else {
+                Some(JsonMap::new())
+            }
```
Comment (tustvold) [inline arrow-json/src/writer.rs:143]:
```suggestion
            array.is_valid(index).then(JsonMap::new)
```

File: arrow-json/src/writer.rs [hunk 2]
```diff
@@ -227,7 +235,17 @@ fn array_to_json_array_internal(
             .collect(),
         DataType::Struct(_) => {
             let jsonmaps = struct_array_to_jsonmap_array(array.as_struct(), explicit_nulls)?;
-            Ok(jsonmaps.into_iter().map(Value::Object).collect())
+            let json_values = jsonmaps
+                .into_iter()
+                .map(|maybe_map| {
+                    if let Some(map) = maybe_map {
```
Comment (tustvold) [inline arrow-json/src/writer.rs:241]: A match might be nicer

**Review by tustvold — APPROVED** [review]
Makes sense to me, just some minor nits

File: arrow-json/src/writer.rs (hunk 2 above)
Comment (Jefffrey) [reply arrow-json/src/writer.rs]: Refactored to use `map` and `unwrap_or`

**Pull request merged**